docs/source/spec/core/stream-traits.rst

```diff
@@ -1,191 +1,100 @@
+once. This includes both streaming binary data and event streams.
 
 .. contents:: Table of contents
     :depth: 2
     :local:
     :backlinks: none
 
+.. _streaming-trait:
 
-.. _single-event-event-stream:
-
---------------------------
-Single-event event streams
---------------------------
-
-A *single-event event stream* is an event stream that streams zero or more
-instances of a specific structure shape.
-
-.. _single-event-input-eventstream:
-
-The following example defines an operation that uses a single-event event
-stream in its input:
-
-.. tabs::
-
-    .. code-tab:: smithy
-
-        namespace smithy.example
+-------------------
+``streaming`` trait
+-------------------
 
-        operation PublishMessages {
-            input: PublishMessagesInput
-        }
+Summary
+    Indicates that the data represented by the shape needs to be streamed.
 
-        structure PublishMessagesInput {
-            room: String,
+    When applied to a blob, this simply means that the data could be very
+    large and thus should not be stored in memory or that the size is unknown
+    at the start of the request.
 
-            @eventStream
-            messages: Message,
-        }
+    When applied to a union, it indicates that shape represents an
+    `event stream <event-streams>`.
+Trait selector::
+    ``:each(blob, union)``
+Value type
+    ``structure``
 
-        structure Message {
-            message: String,
-        }
+The value of the ``streaming`` trait is a structure that supports the following
+optional members:
 
-    .. code-tab:: json
+.. list-table::
+    :header-rows: 1
+    :widths: 10 10 80
 
-        {
-            "smithy": "1.0.0",
-            "shapes": {
-                "smithy.example#PublishMessages": {
-                    "type": "operation",
-                    "input": {
-                        "target": "smithy.example#PublishMessagesInput"
-                    }
-                },
-                "smithy.example#PublishMessagesInput": {
-                    "type": "structure",
-                    "members": {
-                        "room": {
-                            "target": "smithy.api#String"
-                        },
-                        "messages": {
-                            "target": "smithy.example#Message",
-                            "traits": {
-                                "smithy.api#eventStream": true
-                            }
-                        }
-                    }
-                },
-                "smithy.example#Message": {
-                    "type": "structure",
-                    "members": {
-                        "message": {
-                            "target": "smithy.api#String"
-                        }
-                    }
-                }
-            }
-        }
+    * - Property
+      - Type
+      - Description
+    * - requiresLength
+      - ``boolean``
+      - Indicates that the stream must have a known size.
```

Comment:
This can only be set when attached to blobs, right? Should this maybe be a separate trait to make this discrepancy less weird?

Reply:
Maybe. Is there not a use case for it for event streams?

Reply:
I doubt it. Event streams usually don't have a predefined size of bounds.